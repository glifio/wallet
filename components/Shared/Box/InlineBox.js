import {
  color,
  typography,
  border,
  layout,
  space,
  shadow,
  flexbox,
  grid,
  position
} from 'styled-system'
import styled from 'styled-components'

export default styled.span`
    min-width: 0;
    box-sizing: border-box;
    ${position}
    ${border}
    ${space}
    ${color}
    ${layout}
    ${flexbox}
    ${typography}
    ${grid}
    ${shadow}
  `
