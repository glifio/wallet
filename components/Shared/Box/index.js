import {
  color,
  typography,
  border,
  layout,
  space,
  flexbox
} from 'styled-system'
import styled from 'styled-components'

export default styled.div`
  min-width: 0;
  box-sizing: border-box;
  ${border}
  ${space}
  ${color}
  ${layout}
  ${flexbox}
  ${typography}
`
