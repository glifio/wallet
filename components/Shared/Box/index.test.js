import { cleanup, render } from '@testing-library/react'
import Box from '.'

describe('Box', () => {
  afterEach(cleanup)
  test('renders the box', () => {
    const { container } = render(<Box />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders the box with the right attributes', () => {
    const { container } = render(
      <Box
        position='flex'
        flexDirection='row'
        border='1'
        width='auto'
        alignItems='center'
        justifyContent='space-between'
        bg='blue'
        p={4}
      />
    )

    const box = container.querySelector('div')
    expect(box).toHaveStyle('width: auto;')
    expect(box).toHaveStyle('position: flex;')
    expect(box).toHaveStyle('border: 1;')
    expect(box).toHaveStyle('align-items: center;')
    expect(box).toHaveStyle('justify-content: space-between;')
    expect(box).toHaveStyle('background-color: blue;')
    expect(box).toHaveStyle('padding: 32px;')
  })
})
