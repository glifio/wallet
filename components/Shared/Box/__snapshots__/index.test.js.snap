// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders the box 1`] = `
.c0 {
  min-width: 0;
  box-sizing: border-box;
}

<div
  class="c0"
/>
`;
